tripy/tests/frontend/test_cache.py

```diff
@@ -0,0 +1,113 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pytest
+
+import tripy as tp
+
+from tripy.frontend.trace import Trace
+from tripy.frontend.cache import ExecutableCache
+
+
+@pytest.fixture
+def cache():
+    return ExecutableCache()
+
+
+class TestCache:
```

Comment:
Also need a test for what happens if we have an identical computation graph but with different input shapes.